src/test/java/com/exasol/adapter/dialects/postgresql/PostgreSQLSqlDialectIT.java

```diff
@@ -1,622 +1,598 @@
+        final String qualifiedTableName = SCHEMA_POSTGRES_UPPERCASE_TABLE + "." + TABLE_POSTGRES_LOWER_CASE;
+        statementPostgres.execute("CREATE TABLE " + qualifiedTableName + " (x INT, y INT)");
+    }
+
+    private static void createVirtualSchema(final String virtualSchemaName, final String originSchemaName,
+            final Optional<String> additionalParameters) throws SQLException {
+        final StringBuilder builder = new StringBuilder();
+        builder.append("CREATE VIRTUAL SCHEMA ");
+        builder.append(virtualSchemaName);
+        builder.append(" USING " + SCHEMA_EXASOL_TEST + "." + ADAPTER_SCRIPT_EXASOL + " WITH ");
+        builder.append("SQL_DIALECT     = 'POSTGRESQL' ");
+        builder.append("CATALOG_NAME     = '" + postgresqlContainer.getDatabaseName() + "' ");
+        builder.append("CONNECTION_NAME = '" + CONNECTION_POSTGRES_JDBC + "' ");
+        builder.append("SCHEMA_NAME     = '" + originSchemaName + "' ");
+        additionalParameters.ifPresent(builder::append);
+        final String sql = builder.toString();
+        LOGGER.info("Creating virtual schema with query: " + sql);
+        statementExasol.execute(sql);
     }
 
     @Test
     void testSelectSingleColumn() throws SQLException {
-        final String query = String.format("SELECT x FROM %s.t", VIRTUAL_SCHEMA);
-        final ResultSet result = executeQuery(query);
-        assertNextRow(result, 1L);
+        final String qualifiedExpectedTableName = SCHEMA_EXASOL_TEST + ".TABLE_POSTGRES_SIMPLE_EXPECTED";
+        statementExasol.execute("CREATE OR REPLACE TABLE " + qualifiedExpectedTableName + " (x INT)");
+        statementExasol.execute("INSERT INTO " + qualifiedExpectedTableName + " VALUES (1)");
+        final ResultSet expectedResultSet = statementExasol.executeQuery("SELECT * FROM " + qualifiedExpectedTableName);
+        final ResultSet actualResultSet = statementExasol
+                .executeQuery("SELECT * FROM " + VIRTUAL_SCHEMA_POSTGRES + "." + TABLE_POSTGRES_SIMPLE);
+        assertThat(actualResultSet, matchesResultSet(expectedResultSet));
     }
 
     // Join Tests -------------------------------------------------------------
     @Test
     void testInnerJoin() throws SQLException {
-        final String query = "SELECT * FROM  " + VIRTUAL_SCHEMA + ".t1 a INNER JOIN  " + VIRTUAL_SCHEMA
-                + ".t2 b ON a.x=b.x";
-        final ResultSet result = executeQuery(query);
-        assertAll(() -> assertNextRow(result, 2L, "bbb", 2L, "bbb"), () -> assertFalse(result.next()));
+        final ResultSet expected = integrationTestSetupManager.GetSelectAllFromJoinExpectedTable(statementExasol,
+                SCHEMA_EXASOL_TEST, "(x INT, y VARCHAR(100), a INT, b VARCHAR(100))", "VALUES(2,'bbb', 2,'bbb')");
+        final ResultSet actual = statementExasol.executeQuery("SELECT * FROM " + QUALIFIED_TABLE_JOIN_NAME_1
```

Comment:
When I try to read this, the constants make it harder.

I think this is one of those cases where duplication is a good thing. Please remove the constants in those situations.

Reply:
The problem is that If I rename the table, I will have to do it by hands for all the queries... I can replace `QUALIFIED_TABLE_JOIN_NAME_1` with  `VIRTUAL_SCHEMA_POSTGRES + "." + TABLE_JOIN_1` if it helps.